assets/src/dashboard/components/typeahead-input/stories/index.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { action } from '@storybook/addon-actions';
+import { boolean, number, text } from '@storybook/addon-knobs';
+import { useState } from 'react';
+/**
+ * Internal dependencies
+ */
+import TypeaheadInput from '../';
+import { getTypeaheadOptions } from '../../index';
+
+export default {
+  title: 'Dashboard/Components/TypeaheadInput',
+  component: TypeaheadInput,
+};
+
+const items = [
+  { label: 'Pepper', value: 'pepper' },
+  { label: 'Lemon', value: 'lemon' },
+  { label: 'Salt', value: 'salt' },
+  { label: 'Paprika', value: 'paprika' },
+  { label: 'Oregano', value: 'oregano' },
+  { label: 'Lemon Salt', value: 'lemonsalt' },
+  { label: 'Basil', value: 'basil' },
+  { label: 'water', value: 'agua' },
+];
+export const _default = () => {
+  const [value, setValue] = useState('');
+  return (
+    <TypeaheadInput
+      inputId={'demo-search-component'}
+      items={items}
+      onChange={(inputValue) => {
+        if (!inputValue) {
+          setValue('');
+        }
+        action(`input changed ${inputValue}`)(inputValue);
+      }}
+      maxItemsVisible={number('maxItemsVisible', 7)}
+      value={value}
+      placeholder={text('placeholder', 'Search Stories')}
+      ariaLabel={text('ariaLabel', 'my search for seasonings')}
+      disabled={boolean('disabled')}
+    />
+  );
+};
+
+export const _filterLoadedSearchItems = () => {
+  const [value, setValue] = useState('Sports');
+  return (
+    <TypeaheadInput
+      inputId={'demo-search-component'}
+      items={getTypeaheadOptions(items, value)}
```

Comment:
Question, what if instead of using lodash memoize for `getTypeaheadOptions`, you used `useMemo` at this level of the control.  Meaning something like this:
```
const [value, setValue] = useState('Sports');
const filteredItems = useMemo(() => (
  getTypeaheadOptions(items, value)
), [items, value]);
...
```
It would mean we'd have to remember to apply this `useMemo` pattern any time we used `getTypeaheadOptions` for a typeahead control... sooooo, maybe it's not worth it. 🤔

However, if the filtering were moved to within `TypeaheadInput` then it could handle the memoizing itself 🤔

Reply:
I'd love to get away from using lodash here but I'm hesistant to move the `getTypeaheadOptions` into `TypeaheadInput` - it started out there, and then I moved it to a util so that the typeahead doesn't care how the data gets filtered. Since we don't know how the actual "filtering" is getting handled - whether we get the data all at once from the server and then need to filter ourselves (which if that's the case all the time just dropping the util and using `useMemo` would be great). However, right now we also have to be sure we're handling for data that's getting filtered in real time - which would make this util completely obsolete.

Reply:
Yeah I agree, it's hard to know which way would be best without knowing more implementation details.  I'm fine with it as is, and we can refactor later (if needed) once things are hooked up.

Reply:
Sounds good, glad you called this out - now two of us will have it on our radar when a better way to handle filtering comes up!